### PR TITLE
nixVersions.2_3: 2.3.16 -> 2.3.17

### LINKED ISSUE #265916: Update request: nix_2_3 2.3.16 -> 2.3.17
- Package name: nix_2_3
- Latest released version: 2.3.17 (see https://github.com/NixOS/nix/releases/tag/2.3.17)
- Current version on the unstable channel: 2.3.16
- Current version on the stable/release channel: 2.3.16
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@edolstra @lovesegfault @Artturin 

-----

Note for maintainers: Please tag this issue in your PR.

See https://github.com/NixOS/nix/issues/9244#issuecomment-1795041731 for context.

The release script apparently doesn't upload source tarballs anymore, so 2.3.17 sources look more like the other ones.

There's a bunch of `atLeast24` expressions in `pkgs/tools/package-management/nix/common.nix` that might need to be revisited.

### PR DESCRIPTION
**Pull request by RaitoBezarius**
## Description of changes

2.3.17 became a bit more like 2.4, this is reflected in the common expression.

TODO:

- [x] Fix documentation build.
- [x] Add separated maintenance fields, defaulting to prior maintainers of Nix core team?
- [x] Add maintainers for 2.3 series.

Fixes #265916.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/package-management/nix/common.nix [hunk 1]
```diff
@@ -133,12 +131,12 @@ self = stdenv.mkDerivation {
     nlohmann_json
   ];
 
-  NIX_LDFLAGS = lib.optionals (!atLeast24) [
-    # https://github.com/NixOS/nix/commit/3e85c57a6cbf46d5f0fe8a89b368a43abd26daba
-    (lib.optionalString enableStatic "-lssl -lbrotlicommon -lssh2 -lz -lnghttp2 -lcrypto")
-    # https://github.com/NixOS/nix/commits/74b4737d8f0e1922ef5314a158271acf81cd79f8
-    (lib.optionalString (stdenv.hostPlatform.system == "armv5tel-linux" || stdenv.hostPlatform.system == "armv6l-linux") "-latomic")
-  ];
+#  NIX_LDFLAGS = lib.optionals (!atLeast24) [
```
Comment (flokli) [inline pkgs/tools/package-management/nix/common.nix:134]: should be removed?

Comment (RaitoBezarius) [reply pkgs/tools/package-management/nix/common.nix:134]: I feel like so yes but unsure yet.

Comment (flokli) [reply pkgs/tools/package-management/nix/common.nix]: dropped. I checked pkgsStatic does still build, don't have an armv7 machine to test the build there.

File: pkgs/tools/package-management/nix/common.nix [hunk 2]
```diff
@@ -86,13 +92,17 @@ self = stdenv.mkDerivation {
...
   nativeBuildInputs = [
     pkg-config
-  ] ++ lib.optionals atLeast24 [
     autoconf-archive
     autoreconfHook
     bison
     flex
     jq
-  ] ++ lib.optionals (atLeast24 && enableDocumentation) [
+  ] ++ lib.optionals (enableDocumentation && version == "2.3.17") [
```
Comment (sternenseemann) [inline pkgs/tools/package-management/nix/common.nix:100]: `!atLeast24`?

Comment (flokli) [reply pkgs/tools/package-management/nix/common.nix]: Updated, thanks.

**Comment by flokli** [comment]
I don't understand the error that ofborg is reporting, both `nix` and `nix_2_3` attributes build here?!

**Comment by RaitoBezarius** [comment]
@ofborg eval

you mad thing you made me waste my time

**Comment by flokli** [comment]
Ofborg seems green, let's get this in?

**Comment by Artturin** [comment]
@ofborg build nixVersions.nix_2_3

**Pull request merged**